modules/snmp-dest/snmpdest.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017 Balabit
+ * Copyright (c) 2002-2011 BalaBit IT Ltd, Budapest, Hungary
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef SNMPDEST_H_INCLUDED
+#define SNMPDEST_H_INCLUDED
+
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+
+#include "driver.h"
+#include "mainloop-worker.h"
+
+typedef struct
+{
+  LogDestDriver super;
+
+  gchar *version;
+  gchar *host;
+  GString *host_port;
+  gint port;
+  GList *snmp_objs; /* the values are stored as string triplets (oid, type, value) - the list size is always dividable by 3 */
```

Comment:
A nice refactor would be to create a structure for the triplet.

Reply:
I'll do this refactor alongside the threaded destination work